tools/failures.py

```diff
@@ -31,9 +32,12 @@ def get_raw_data(start_date, end_date):
 
     url = "http://alertmanager.allizom.org/data/seta/?startDate=%s&endDate=%s" % \
           (start_date.strftime("%Y-%m-%d"), end_date.strftime("%Y-%m-%d"))
-
-    response = retry(requests.get, args=(url, ),
-                     kwargs={'headers': headers, 'verify': True})
+    try:
+        response = retry(requests.get, args=(url, ),
+                         kwargs={'headers': headers, 'verify': True})
+    except Exception as e:
+        logger.debug("The request to %s is failed due to %s" % (url, e))
+        return
```

Comment:
correct, lets see where else we can make things more robust!

On Wed, Sep 7, 2016 at 11:18 AM, Tiramisu 1993 <notifications@github.com>
wrote:

> In tools/failures.py
> <https://github.com/mozilla/ouija/pull/136#discussion_r77843718>:
>
> > @@ -31,9 +32,12 @@ def get_raw_data(start_date, end_date):
> >
> >      url = "http://alertmanager.allizom.org/data/seta/?startDate=%s&endDate=%s" % \
> >            (start_date.strftime("%Y-%m-%d"), end_date.strftime("%Y-%m-%d"))
> > -
> > -    response = retry(requests.get, args=(url, ),
> > -                     kwargs={'headers': headers, 'verify': True})
> > +    try:
> > +        response = retry(requests.get, args=(url, ),
> > +                         kwargs={'headers': headers, 'verify': True})
> > +    except Exception as e:
> > +        logger.debug("The request to %s is failed due to %s" % (url, e))
> > +        return
>
> alright, I think I understand what's your idea in here. We should keep
> seta running even something wrong on the network, right? I think return a
> {} is a good idea and I will look after the other places :)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/ouija/pull/136/files/3a562384c0013f1d28351614eb0aeac0114c3e18#r77843718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD0fCccIdPuiHtMFJdgZCmzuf3UztPOks5qntXJgaJpZM4J215G>
> .
>